example/newrepo/main.go

```diff
@@ -11,30 +11,36 @@ package main
 
 import (
 	"context"
+	"flag"
 	"fmt"
 	"log"
 	"os"
-	"strings"
 
 	"github.com/google/go-github/github"
 	"golang.org/x/oauth2"
 )
 
+var name = flag.String("name", "", "Name of repo to create in authenticated user's GitHub account.")
+var description = flag.String("description", "", "Description of created repo.")
+var private = flag.Bool("private", false, "Will created repo be private.")
+
 func main() {
+	flag.Parse()
 	token := os.Getenv("GITHUB_AUTH_TOKEN")
-	name := strings.Join(os.Args[1:], "-")
 	if token == "" {
 		log.Fatal("Unauthorized: No token present")
 	}
-	if name == "" {
+	if *name == "" {
 		log.Fatal("No name: New repos must be given a name")
 	}
 	ctx := context.Background()
 	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
 	tc := oauth2.NewClient(ctx, ts)
 	client := github.NewClient(tc)
-	r := &github.Repository{Name: github.String(name)}
 
+	r := &github.Repository{Name: github.String(*name)}
```

Comment:
Since the flags are already pointers, `github.String` and `github.Bool` in these 3 lines don't need to be used at all.

```go
r := &github.Repository{Name: name, Private: private, Description: description}
```